kpatch-build/kpatch-build

```diff
@@ -592,10 +617,6 @@ fi
 # unsupported kernel option checking: CONFIG_DEBUG_INFO_SPLIT
 grep -q "CONFIG_DEBUG_INFO_SPLIT=y" "$CONFIGFILE" && die "kernel option 'CONFIG_DEBUG_INFO_SPLIT' not supported"
 
-echo "Testing patch file"
-cd "$SRCDIR" || die
-patch -N -p1 --dry-run < "$PATCHFILE" || die "source patch file failed to apply"
-cp "$PATCHFILE" "$APPLIEDPATCHFILE" || die
```

Comment:
It would be nice to keep this test.

What if we have an apply_patches() function along with a remove_patches() function.  They could use a global numeric variable to track the number of patches currently applied.  apply_patches() and remove_patches() could be used here, and then apply_patches() could be used again in the "Building original kernel" section and remove_patches() could be used again in the cleanup.

Then we could get rid of APPLIEDPATCHFILE, as well as the SRC_LIST stuff.  Thoughts?

Reply:
Yeah, the `APPLIEDPATCHFILE` thing is a little wart, so creating an `apply_patches()` function with some kind of memory of which ones were applied sounds nicer.